dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/FoundDimensionItemValue.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2004-2021, University of Oslo
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * Neither the name of the HISP project nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.common;
+
+import lombok.Data;
+
+import org.hisp.dhis.category.CategoryOptionCombo;
+import org.hisp.dhis.organisationunit.OrganisationUnit;
+import org.hisp.dhis.period.Period;
+
+/**
+ * DimensionItemObject data as found with organisation unit, period, attribute
+ * option combination, and a value.
+ *
+ * @author Jim Grace
+ */
+@Data
```

Comment:
`FoundDimensionItemValue` needs `equals` because `PredictionDataValueFetcherTest` tests against these objects. And it needs `toString` so it will print something helpful if the test fails:

>         assertContainsOnly( data1.getValues(), foundValueA, foundValueC, foundValueD, foundValueE );

I've changed `@Data` to `@Getter`, `@ToString`, `@EqualsAndHashCode`, and `@AllArgsConstructor`. Is this preferable?

Reply:
Yes. `@ToString`, `@EqualsAndHashCode` also have the `@ToString( onlyExplicitlyIncluded = true )` and `@ToString.Include` to get a fixed set of fields included. This is the "safest" way to auto-generate such methods with lombok.